### PR TITLE
BUG: fix to_latex bold_rows option

### LINKED ISSUE #16707: DataFrame.to_latex(): bold_rows option is disregarded
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
df = pd.DataFrame({'acol':['a', 'aa'], 'bcol':['b', 'bb']}, index=['row1', 'row2'])
print df.to_latex(bold_rows=True)
```
```
\begin{tabular}{lll}
\toprule
{} & acol & bcol \\
\midrule
row1 &    a &    b \\
row2 &   aa &   bb \\
\bottomrule
\end{tabular}
```
#### Problem description
Despite `bold_rows` is documented to be an option of `to_latex()` and to be `True` by default, in reality, this option has no effect. Moreover, the current behaviour corresponds with what would be expected for `bold_rows=False` and is thus in contrast with the documented default 'bold_rows=True' behaviour.

#### Expected Output
```
\begin{tabular}{lll}
\toprule
{} & acol & bcol \\
\midrule
\textbf{row1} &    a &    b \\
\textbf{row2} &   aa &   bb \\
\bottomrule
\end{tabular}
```
#### Output of ``pd.show_versions()``

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.9.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-78-generic
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.2
pytest: None
pip: 9.0.1
setuptools: 36.0.1
Cython: 0.25.2
numpy: 1.13.0
scipy: 0.19.0
xarray: None
IPython: 5.4.1
sphinx: None
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: None
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.8
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: None
lxml: 3.8.0
bs4: 4.6.0
html5lib: 0.999999999
sqlalchemy: 1.1.10
pymysql: 0.7.11.None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None

### PR DESCRIPTION
**Pull request by kdebrab**
- [x] closes #16707
 - [x] tests added / passed
 - [ ] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/io/formats/format.py [hunk 1]
```diff
@@ -943,6 +943,11 @@ def get_col_type(dtype):
                          if x else '{}') for x in row]
             else:
                 crow = [x if x else '{}' for x in row]
+            if self.fmt.kwds.get('bold_rows', True) and self.fmt.index:
```
Comment (jreback) [inline pandas/io/formats/format.py]: in the constructor do a ``self.bold_rows = self.fmt.kwd.get('bold_rows', True)``; see ``HTMLFormatter``. 

Then just use this here

**Review by jreback — CHANGES_REQUESTED** [review]
looks reasonable. pls add a bug fix whatsnew entry (in IO) for 0.20.3

**Comment by kdebrab** [comment]
I had to adapt  the expected result of all existing tests where. to_latex() was used, as the default behaviour changed. I'm not so sure though whether it's a good idea to set bold_rows to True by default. I guess the reason for this documented  default is the analogy with. to_html(), but in that output also the column headers are bold. In addition, bold row and/or column headers do not seem very common in latex community examples.

**Comment by chris-b1** [comment]
Could you show an example what the before/after output looks like rendered?  I'm inclined to agree that changing the default isn't a good idea.

**Comment by kdebrab** [comment]
With `bold_rows=False` (i.e. 'before' output):
image
With `bold_rows=True` (would be 'after' default output):
image

It's a matter of taste, but I don't like the latter one that much...

I'm in favor of keeping the default output the same as before, and correcting the signature (and docstring) of .to_latex() accordingly.

**Comment by TomAugspurger** [comment]
> I'm in favor of keeping the default output the same as before, and correcting the signature (and docstring) of .to_latex() accordingly.

+1

**Comment by kdebrab** [comment]
@jreback: As proposed, I made sure to have the same default behaviour as before (i.e. `bold_rows=False`) and corrected the `.to_latex()` signature accordingly.

I also added and updated a whatsnew entry as you asked, so I guess this is ready to be merged.

**Comment by jreback** [comment]
lgtm. can you rebase. ping on green. @TomAugspurger

File: doc/source/whatsnew/v0.20.3.txt [hunk 2]
```diff
@@ -55,7 +54,8 @@ Indexing
 I/O
 ^^^
 
--- Bug in ``pd.read_csv()`` in which files weren't opened as binary files by the C engine on Windows, causing EOF characters mid-field, which would fail (:issue:`16039`, :issue:`16559`, :issue`16675`)
+- Bug in ``pd.read_csv()`` in which files weren't opened as binary files by the C engine on Windows, causing EOF characters mid-field, which would fail (:issue:`16039`, :issue:`16559`, :issue`16675`)
+- Bug in ``.to_latex()`` where ``bold_rows`` was wrongly specified to be True by default, whereas in reality row labels remained non-bold whatever parameter provided. (:issue:`16707`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.20.3.txt]: Maybe just say here that the default behavior is the same (not bold), but can now be overridden correctly.

**Review by TomAugspurger — APPROVED** [review]
Looks good, small suggestion to clarify the whatsnew that you can change as you rebase.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**